### PR TITLE
Feature/103 search results

### PR DESCRIPTION
**Pull request by davideastmond**
Draft: work in progress.,

- Functionality for querying a user (we can query firstName, lastName and jobTitle fields)
- Functionality for querying a main public and private thread (`content.html` and `content.hashTags`).
- Updated user and thread schema to create a text index.
- Started a `/search` route. Need to discuss query structure.
- Clean up `createTestUsers` function calls to be more succinct (there were unnecessary arguments being provided to the functions)

- Will start working on querying threadComments associated with public and private threads.

### CODE REVIEW

File: src/db/search/helpers/does-match.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const stringWithAllCharactersRemoved = stripStringOfAllNonAlphaCharacters(data.input);
+
+  const query = stringWithAllCharactersRemoved.toLowerCase().trim().split(" ");
+  for (let i = 0; i < flattenedData.length; i++) {
+    if (query.indexOf(flattenedData[i].toLowerCase()) >= 0 ) {
+      return true;
+    }
+  }
+  return false;
+}
```
Comment (davideastmond) [inline src/db/search/helpers/does-match.ts:18]: not sure we'll need these functions

File: src/db/search/helpers/strip-characters.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function stripStringOfAllNonAlphaCharacters(input: string) {
+  if (input) {
+    return input.replace(/[^\w\s]|_/g, "")
+    .replace(/\s+/g, " ");
+  }
+
+  return "";
```
Comment (davideastmond) [inline src/db/search/helpers/strip-characters.ts:7]: not sure we'll need this

File: src/db/search/helpers/does-match.ts (hunk 1 above)
Comment (lily-law) [reply src/db/search/helpers/does-match.ts:18]: Maybe comment them out until they're used or removed... that way if we forget about them we won't go searching for where they're used

**Comment by lily-law** [comment]
:rocket: looking good so far

File: src/db/search/helpers/does-match.ts (hunk 1 above)
Comment (davideastmond) [reply src/db/search/helpers/does-match.ts]: I deleted these functions.

**Comment by lily-law** [comment]
:cool: Looking good. Lets look at adding a limit() and skip() for the points where we run .find() on the db to speed up searching and reduce load to this

**Review by lily-law — COMMENTED** [review]
:+1:

**Comment by davideastmond** [comment]
> 🆒 Looking good. Lets look at adding a limit() and skip() for the points where we run .find() on the db to speed up searching and reduce load to this
Good idea. I added this to the PR

**Pull request merged**